Gems/Atom/RPI/Code/Source/RPI.Public/Model/ModelLod.cpp

```diff
@@ -344,6 +344,95 @@ namespace AZ
+            UvStreamTangentBitmask* uvStreamTangentBitmaskOut,
+            const ShaderInputContract& contract,
+            size_t meshIndex,
+            const MaterialModelUvOverrideMap& materialModelUvMap,
+            const MaterialUvNameMap& materialUvNameMap) const
+        {
+            streamBufferViewsOut.clear();
+
+            RHI::InputStreamLayoutBuilder layoutBuilder;
+
+            const Mesh& mesh = m_meshes[meshIndex];
+
+            bool success = true;
+
+                   // Searching for the first UV in the mesh, so it can be used to paired with tangent/bitangent stream
+            auto firstUv = FindFirstUvStreamFromMesh(meshIndex);
+            auto defaultUv = FindDefaultUvStream(meshIndex, materialUvNameMap);
+            if (uvStreamTangentBitmaskOut)
+            {
+                uvStreamTangentBitmaskOut->Reset();
+            }
+
+            for (auto& contractStreamChannel : contract.m_streamChannels)
+            {
+                auto iter = FindMatchingStream(meshIndex, materialModelUvMap, materialUvNameMap, contractStreamChannel, defaultUv, firstUv, uvStreamTangentBitmaskOut);
+
+                if (iter == mesh.m_streamInfo.end())
+                {
+                    if (contractStreamChannel.m_isOptional)
+                    {
+                        // The configuration of the dummy input stream is a bit touchy, and could result in crashes or validation failures that are platform-specific.
+                        // If you modify this code, be sure to test with "-rhi-device-validation=enable" on every platform.
+                        // Here are some criteria that we've noticed before, even for empty buffers:
+                        // Metal: Mesh stream formats need to be at least 4 byte aligned.
+                        // Vulkan: Mesh stream data type (float vs uint) must match the shader, or validation errors are reported
+                        //        ("does not match vertex shader input type")
+                        // Vulkan: We can't just use a null buffer pointer because vulkan will occasionally crash. So we bind some valid non-null buffer and view it with length 0.
+                        // Dx12: Mesh stream data type (float vs uint) must match the shader, or validation warnings are reported
+                        //       ("the matching entry in the Input Layout declaration ... specifies mismatched format").
+                        //
+                        // The stride value does not seem to matter, just the Format type. Still, we use GetFormatSize to set an accurate stride.
+
+                        RHI::Format dummyStreamFormat = RHI::Format::R32G32B32A32_FLOAT;
+                        layoutBuilder.AddBuffer()->Channel(contractStreamChannel.m_semantic, dummyStreamFormat);
+                        RHI::MultiDeviceStreamBufferView dummyBuffer{*mesh.m_indexBufferView.GetBuffer(), 0, 0, RHI::GetFormatSize(dummyStreamFormat)};
```

Comment:
A lot of this is copy paste from the other GetStreamsForMesh. Consider creating a common function to avoid duplication

Reply:
This will be removed in the final rewrite and is only temporarily necessary.